eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -88,6 +90,35 @@ object ReplaceableTxFunder {
     addSigs(unsignedCommitTx, PlaceHolderPubKey, PlaceHolderPubKey, PlaceHolderSig, PlaceHolderSig)
   }
 
+  /**
+   * The on-chain feerate can be arbitrarily high, but it wouldn't make sense to pay more fees than the amount we're
+   * trying to claim on-chain. We compute how much funds we have at risk and the feerate that matches this amount.
+   */
+  def maxFeerate(txInfo: ReplaceableTransactionWithInputInfo, commitment: FullCommitment): FeeratePerKw = {
+    // We don't want to pay more in fees than the amount at risk in untrimmed pending HTLCs.
+    val maxFee = txInfo match {
+      case tx: HtlcTx => tx.input.txOut.amount
+      case tx: ClaimHtlcTx => tx.input.txOut.amount
+      case _: ClaimLocalAnchorOutputTx =>
+        val htlcBalance = commitment.localCommit.htlcTxsAndRemoteSigs.map(_.htlcTx.input.txOut.amount).sum
+        val mainBalance = commitment.localCommit.spec.toLocal.truncateToSatoshi
+        // If there are no HTLCs or a low HTLC amount, we may still want to get back our main balance.
+        // In that case, we spend at most 5% of our balance in fees.
+        htlcBalance.max(mainBalance * 5 / 100)
+    }
+    // We cannot know beforehand how many wallet inputs will be added, but an estimation should be good enough.
+    val weight = txInfo match {
+      // For HTLC transactions, we add a p2wpkh input and a p2wpkh change output.
```

Comment:
I don't understand the comment and I don't think it matches the implementation. Why wouldn't all replaced transactions have an additional p2wpkh input?

Reply:
The claim-htlc-txs (when the _remote_ commit is confirmed) don't need any additional input, we simply lower their output to RBF them (which is nice because we don't need the bitcoind wallet and it automatically caps the fee to the HTLC amount). It's only the htlc-txs and the anchor transaction that add an input/output pair.